### PR TITLE
Make notEqual work on relations 

### PR DESCRIPTION
**Pull request by flovilmart**
Adds support for $eq, $ne and $nin when querying on relation keys

### CODE REVIEW

**Comment by drew-gross** [comment]
This looks like it's going to not work if you use mix and match equalTo, notEqualtTo, containedIn, and notContainedIn all in the same query. Have you checked for that?

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -452,14 +452,25 @@ DatabaseController.prototype.reduceInRelation = function(className, query, schem
...
+      } else if (query[key]['$ne']) {
+        isNegation = true;
+        relatedIds = [query[key]['$ne'].objectId];
       } else {
         relatedIds = [query[key].objectId];
       }
       return this.owningIds(className, key, relatedIds).then((ids) => {
         delete query[key];
-        this.addInObjectIdsIds(ids, query);
+        if (isNegation) {
```
Comment (flovilmart) [inline src/Controllers/DatabaseController.js]: apply negation if needed

**Comment by flovilmart** [comment]
the test case covers all those cases, actually we just fetch the [owningIds](https://github.com/ParsePlatform/parse-server/pull/1350/files#diff-26b0f888e8a20f22d9728a26272f6cddR456), then we apply inclusion or exclusion (https://github.com/ParsePlatform/parse-server/pull/1350/files#diff-26b0f888e8a20f22d9728a26272f6cddR469)

**Comment by flovilmart** [comment]
@drew-gross yes you're definitely right, I didn't check for that.

**Comment by lolobosse** [comment]
Hey,

@drew-gross yes, it was another consequences of my further tests (that I just finished now (you were quicker)), the server do not like the combinations as well. Do you need some further test cases to fix that quicker?

When do you think it could go live?

**Comment by drew-gross** [comment]
@lolobosse I think @flovilmart is already working on this, but If you posted a test case similar to the one in this PR, he would probably find that helpful!

**Comment by flovilmart** [comment]
@drew-gross I posted the support, let me know what you think

**Comment by drew-gross** [comment]
This looks like it should do the trick :100:

**Pull request merged**